block-layer/codecs/dag-json.md

```diff
@@ -42,13 +51,49 @@ The Bytes kind is represented as an object with `"bytes"` as key and a [Multibas
 {"/": { "bytes": String /* Multibase Base64 encoded binary */ }}
 ```
 
-#### Link kind
+### Links
 
-A link is represented as a base encoded CID. CIDv0 and CIDv1 are encoded differently.
+A Link kind is represented as a base encoded CID. CIDv0 and CIDv1 are encoded differently.
 
  - CIDv1 is represented as a Multibase Base32 encoded string. The Base32 encoding is the one described in [RFC 4648, section 6](https://tools.ietf.org/html/rfc4648#section-6) without padding, hence the Multibase prefix is `b`.
  - CIDv0 is represented in its only possible Base58 encoding. The Base58 encoding is the one described in [Base58 draft](https://tools.ietf.org/html/draft-msporny-base58).
 
 ```javascript
 {"/": String /* Base58 encoded CIDv0 or Multibase Base32 encoded CIDv1 */}
 ```
+
+### The Reserved Namespace
+
+Maps with the first key of `"/"` are considered the **reserved namespace** in DAG-JSON as they are used to represent Bytes and Links. There are special rules that restrict certain data forms from being properly encoded in DAG-JSON. These rules allow for the clean representation of Bytes and Links as well as efficient operation of tokenizing decoders. A tokenizing decoder should not need to buffer and back-track more than 4 tokens upon detection of a map that is not properly encoding Links or Bytes.
+
+Specifically, data with the following forms are strictly not valid DAG-JSON and should be rejected by encoders and decoders:
+
+ * Maps with more than one key, where the first key is `"/"` and its value is a string. e.g. `{"/":"foo","bar":"baz"}`.
+   - Where a key exists that sorts before `"/"`, the map is valid, e.g. `{"0bar":"baz","/":"foo"}`.
+   - Where the value of the `"/"` entry is not a string, the map is valid, e.g. `{"/":true,"bar":"baz"}`
```

Comment:
It took me awhile to grasp that the main item is the rule to reject something and that the sub-items are rules that would not be rejected.

Perhaps it could be expended and we give the rules even names. Example:

### Multiple keys

Rejection criteria: Maps with more than one key, where the first key is `"/"` and its value is a string. 

Examples for invald DAG-JSON:
 - `{"/":"foo","bar":"baz"}`.
 - `{"/":"foo","bar":true}`.

Examples for valid DAG-JSON:
 - `{"0bar":"baz","/":"foo"}`: There is a key that sorts before `"/"`. The value following the `"/"` will *not* be interpreted as link.
 - `{"/":true,"bar":"baz"}`: There value after the `"/"` is not a string.
 - `{"/":"foo"}: There is only a the single map key `"/"`. The value is expected to be a valid base encoded CID.

Reply:
I think I've addressed this, but worth a review